task/control-plane-agent/src/main.rs

```diff
@@ -585,26 +589,25 @@ impl NetHandler {
 }
 
 fn sp_port_from_udp_metadata(meta: &UdpMetadata) -> SpPort {
-    use task_net_api::VLAN_RANGE;
-    assert!(VLAN_RANGE.contains(&meta.vid));
-    assert_eq!(VLAN_RANGE.len(), 2);
-
-    match meta.vid - VLAN_RANGE.start {
-        0 => SpPort::One,
-        1 => SpPort::Two,
-        _ => unreachable!(),
+    use task_net_api::VLANS;
+    match VLANS.iter().find(|v| v.vid == meta.vid).unwrap().port {
+        task_net_api::SpPort::One => SpPort::One,
+        task_net_api::SpPort::Two => SpPort::Two,
     }
 }
 
 #[allow(dead_code)]
 fn vlan_id_from_sp_port(port: SpPort) -> u16 {
-    use task_net_api::VLAN_RANGE;
-    assert_eq!(VLAN_RANGE.len(), 2);
-
-    match port {
-        SpPort::One => VLAN_RANGE.start,
-        SpPort::Two => VLAN_RANGE.start + 1,
-    }
+    // TODO should we remove this and just use `Sender` everywhere?
```

Comment:
Maaaybe? I think this is primarily used in the installinator / host phase 2 via MGS step, where the SP is initiating communication by sending to a multicast address that MGS is bound to. I wouldn't object to removing this but it might be kinda fiddly.

Reply:
Yeah, it ended up being fiddly because we swap between `SpPort::One/Two`, which is harder to do with `Sender` types; I've removed that TODO and inlined this whole function into `packet_to_mgs` (because that's the only caller at this point).

See 0c36fb5bd538